lsp/nls/src/config.rs

```diff
@@ -0,0 +1,72 @@
+//! Configuration for the nls server
+use serde::{Deserialize, Serialize};
+
+use std::time::Duration;
+
+fn default_eval_timeout() -> Duration {
+    Duration::from_secs(1)
+}
+fn default_recursion_limit() -> usize {
+    128
+}
+fn default_blacklist_duration() -> Duration {
```

Comment:
Are those used anywhere else? It's usually more idiomatic to just inline the value in the `Default` implementation of the structure holding them.

Reply:
They are (afaik) needed for serde's `default` pragma – which requires a function name if the `Default` instance for the field can't be used. Maybe there's a way around that (it's indeed quite ugly), but I couldn't find it.

Reply:
I believe you can slap `#[serde(default)]` on the whole container and it will use its `Default` impl. Field defaults are used only when you want to have default values for some fields, but not all.

Reply:
Oooooh, indeed. Let me simplify that a bit then

Reply:
Changed in 7926278d. That's indeed much nicer